src/core/sorted_map.cc

```diff
@@ -783,5 +783,16 @@ bool SortedMap::DefragIfNeeded(float ratio) {
   return reallocated;
 }
 
+std::optional<SortedMap::RankAndScore> SortedMap::GetRankAndScore(sds ele, bool reverse) const {
+  ScoreSds obj = score_map->FindObj(ele);
+  if (obj == nullptr)
+    return std::nullopt;
+
+  optional rank = score_tree->GetRank(obj);
```

Comment:
you can pass `reverse` into `GetRank` to avoid copying the reversing logic below.
Similarly DCHECK is redundant as is it is checked by GetRank

Reply:
Yep, I've added the 'reverse' param to ScoreTree::GetRank